File: app/src/main/java/com/wegielek/signalychinese/views/CameraActivity.java
package com.wegielek.signalychinese.views;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

import com.wegielek.signalychinese.R;

public class CameraActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_camera);
    }

    @Override
    public void finish() {
        super.finish();
        overridePendingTransition(R.anim.slide_in_left, R.anim.slide_out_right);
    }
}

File: app/src/main/java/com/wegielek/signalychinese/views/CameraActivity.kt
package com.wegielek.signalychinese.views

import android.os.Build
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.wegielek.signalychinese.R

class CameraActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_camera)
    }

    override fun finish() {
        super.finish()
        if (Build.VERSION.SDK_INT >= 34) {
            overrideActivityTransition(
                OVERRIDE_TRANSITION_CLOSE,
                R.anim.slide_in_left,
                R.anim.slide_out_right
            )
        } else {
            overridePendingTransition(R.anim.slide_in_left, R.anim.slide_out_right)
        }
    }
}